[202405] Add Cumulative CodeQL changes.

apop5 opened the pull request:
## Description

Cumulative CodeQL fixes for the release 202405 Branch.

555b935ce8
a3fa7c62ee
c7e6a02732
5980a5c946
fba1e58c54
d76a41b050

- [x] Impacts functionality?
- [ ] Impacts security?
- [ ] Breaking change?
- [ ] Includes tests?
- [ ] Includes documentation?

## How This Was Tested
Local CI

## Integration Instructions
N/A

corthon committed:
FatPkg: CodeQL Fixes.

Makes changes to comply with alerts raised by CodeQL.

The issues here fall into the following category:

1. Potential use of uninitialized pointer.

TaylorBeebe committed:
FmpDevicePkg: CodeQL Fixes.

Makes changes to comply with alerts raised by CodeQL.

Most of the issues here fall into the following two categories:

1. Potential use of uninitialized pointer.
2. Inconsistent integer width in comparison.

os-d commented on SecurityPkg/Library/DxeTcg2PhysicalPresenceMinimumLib/DxeTcg2PhysicalPresenceMinimumLib.inf:
```diff
@@ -9,7 +9,7 @@
 
-#Override : 00000002 | SecurityPkg/Library/DxeTcg2PhysicalPresenceLib/DxeTcg2PhysicalPresenceLib.inf | 6205753f2abf9126f2711c5f154f9f99 | 2024-07-24T18-06-30 | 69ff4b6fc889c8e66101cad3dcf8be3d516e038d
+#Override : 00000002 | SecurityPkg/Library/DxeTcg2PhysicalPresenceLib/DxeTcg2PhysicalPresenceLib.inf | 0e330f45cfc28a0ec3ac6fccf5985f97 | 2024-07-30T05-15-01 | 185028e6eef7f50c471e2ca0b6a31a40ce408d6e
 # This is not a true override, but spell changes to ensure mu_tiano_plus passes CI is required and changes the hash.
```
Why do we treat this as an override instead of just a spelling update?

apop5 replied:
Removing the comment. It is a true derivative of the original file.

os-d commented on SecurityPkg/SecurityPkg.dsc:
```diff
@@ -80,6 +80,7 @@
   VariablePolicyHelperLib|MdeModulePkg/Library/VariablePolicyHelperLib/VariablePolicyHelperLib.inf
   OemTpm2InitLib|SecurityPkg/Library/OemTpm2InitLibNull/OemTpm2InitLib.inf               ## MU_CHANGE
   Hash2CryptoLib|SecurityPkg/Library/BaseHash2CryptoLibNull/BaseHash2CryptoLibNull.inf   ## MU_CHANGE
+  SafeIntLib |MdePkg/Library/BaseSafeIntLib/BaseSafeIntLib.inf ## MU_CHANGE - CodeQL Change
```
I may be missing it, but I'm not seeing where SafeIntLib is used

apop5 replied:
Removing in latest push.

makubacki committed:
EmbeddedPkg: CodeQL Fixes.

Makes changes to comply with alerts raised by CodeQL.

Most of the issues here fall into the following two categories:

1. Potential use of uninitialized pointer.
2. Inconsistent integer width in comparison.

Co-authored-by: Taylor Beebe <31827475+TaylorBeebe@users.noreply.github.com>
Co-authored-by: kenlautner <85201046+kenlautner@users.noreply.github.com>
Co-authored-by: Bret Barkelew <bret@corthon.com>

makubacki committed:
SecurityPkg: CodeQL Fixes.

Makes changes to comply with alerts raised by CodeQL.

Most of the issues here fall into the following two categories:

1. Potential use of uninitialized pointer.
2. Inconsistent integer width in comparison.

Co-authored-by: Taylor Beebe <31827475+TaylorBeebe@users.noreply.github.com>
Co-authored-by: kenlautner <85201046+kenlautner@users.noreply.github.com>
Co-authored-by: Bret Barkelew <bret@corthon.com>
